tests/unittest.cpp

```diff
@@ -543,14 +545,74 @@ void checkCollections(const ExampleHitCollection& hits, const ExampleClusterColl
     const auto counts = vec.count();
     REQUIRE(counts.size() == 2);
     REQUIRE(counts[0] == i);
-    REQUIRE(counts[1] == i * 2);
+    REQUIRE(counts[1] == i + 42);
     i++;
   }
 
   i = 0;
   for (const auto v : userData) {
     REQUIRE(v == 3.14f * i++);
   }
+
+  // Also check that the buffers we get are valid and have the expected contents
+  hits.prepareForWrite();
+  auto hitBuffers = hits.getBuffers();
+  if (!hits.isSubsetCollection()) {
+    auto* hitPODBuffers = hitBuffers.dataAsVector<ExampleHitData>();
+    REQUIRE(hitPODBuffers->size() == nElements);
+    for (size_t iHit = 0; iHit < nElements; ++iHit) {
+      const auto& hitPOD = (*hitPODBuffers)[iHit];
+      REQUIRE(hitPOD.cellID == static_cast<unsigned long long>(iHit));
+      REQUIRE(hitPOD.energy == 100.f * iHit);
+    }
+  } else {
+    // NOTE: Just basic tests here really. Checking the contents as well here
+    // requires information that is not easily available at the moment
+    REQUIRE(hitBuffers.data == nullptr);
+    REQUIRE(hitBuffers.references->size() == 1);
+    REQUIRE(hitBuffers.vectorMembers->empty());
```

Comment:
This is always true for subset collections and already checked in the `"Subset collection basics"` test case.